components/core/DataView.js

```diff
@@ -248,20 +221,44 @@ export default class DataView extends React.Component {
     window.removeEventListener("remote-data-deletion", this._handleDataDeletion);
     window.removeEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
     window.removeEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
+    window.removeEventListener("scroll", this._handleCheckScroll);
     window.removeEventListener("remote-update-carousel", this._handleUpdate);
   }
 
-  _increment = (direction) => {
-    if (
-      direction > 0 &&
-      this.state.startIndex + VIEW_LIMIT < this.props.viewer.library[0].children.length
-    ) {
-      this.setState({ startIndex: this.state.startIndex + VIEW_LIMIT });
-    } else if (direction < 0 && this.state.startIndex - VIEW_LIMIT >= 0) {
-      this.setState({ startIndex: this.state.startIndex - VIEW_LIMIT });
+  _handleDebounce = (func, wait) => {
+    let timeout;
+
+    return function executedFunction(...args) {
+      const later = () => {
+        clearTimeout(timeout);
+        func(...args);
+      };
+
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+    };
+  };
+
+  _handleScroll = (e) => {
+    const windowHeight =
+      "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;
+    const body = document.body;
+    const html = document.documentElement;
+    const docHeight = Math.max(
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const windowBottom = windowHeight + window.pageYOffset;
+    if (windowBottom >= docHeight) {
```

Comment:
Maybe play around with how many load at a time. I think 10 is too few, and feels very jerky when you're trying to scroll down repeatedly. Load at least enough to fill the height of the screen each time. (Loading will also look better with the new blur loading thing I implemented, so we won't have just grey boxes while images are waiting to load). Potentially also increase the initial starting amount to something higher than 20.

Reply:
You could also allow it to load even when you aren't at the very bottom (like maybe when windowBottom is 100px--which is approx the height of each line--away from docHeight), that would also help avoid the jump that happens when you load more in some cases.

Reply:
I really like this suggestion

Reply:
That makes so much more sense. Then the loading can both start before they get to the true bottom and it will feel more infinite as you scroll. 🙌🏻